src/datahike/connector.cljc

```diff
@@ -23,13 +23,14 @@
         ;; flush for in-memory backends would only make sense if multiple processes access the db
         ;; TODO: update can also be skipped for external stores when single process writes unless memory should be freed up
         ;;       -> add options to config
-        eavt-flushed (cond-> eavt (not= :mem (-> config :store :backend)) (di/-flush backend))
-        aevt-flushed (cond-> aevt (not= :mem (-> config :store :backend)) (di/-flush backend))
-        avet-flushed (cond-> avet (not= :mem (-> config :store :backend)) (di/-flush backend))
+        backend? (not= :mem (-> config :store :backend))
```

Comment:
I would suggest to push this logic into `di/-flush` instead of manually dispatching around its callsites.

Reply:
If it's important to you, you are free to change it accordingly.